test/sockets.test.js

```diff
@@ -1,63 +1,186 @@
 import 'regenerator-runtime/runtime';
 
 import Socket from '../src/sockets';
-import Logger from '../src/logger';
+import { Logger, SOCKET_PING } from 'syft-helpers.js';
 
 import { WebSocket, Server } from 'mock-socket';
 
 global.WebSocket = WebSocket;
 
 const url = 'ws://localhost:8080/';
 
-const onMessage = event => console.log('NEW MESSAGE', event);
-const onOpen = event => console.log('NEW OPEN', event);
-const onClose = event => console.log('NEW CLOSE', event);
-const onError = event => console.log('NEW ERROR', event);
+/**
+ * Creates promise from the first event call
+ * @param emitter Object that emits the event
+ * @param event Event name
+ * @returns {Promise}
+ */
+function makeEventPromise(emitter, event) {
+  let resolver;
+  let promise = new Promise(resolve => resolver = resolve);
+  emitter.on(event, data => resolver(data));
+  return promise;
+}
 
 describe('Sockets', () => {
+  // common test objects
+  let mockServer;
+
   jest.spyOn(console, 'log');
 
+  beforeEach(() => {
+    mockServer = new Server(url);
+    mockServer.connected = makeEventPromise(mockServer, 'connection');
+  });
+
   afterEach(() => {
+    mockServer.close();
     jest.clearAllMocks();
   });
 
-  test('can communicate with a socket server', async () => {
-    const message = { message: 'This is a test' };
+  test('keep-alive messages', async() => {
```

Comment:
Test messages (`'keep-alive messages'`) should read as a full sentence with the description.  Our description (written above) is `'Sockets'`, so a full sentence should be `'Sockets can have a custom keep-alive timeout'` with `'Sockets'` being the description message (from the `describe()` function) and `'can have a custom keep-alive timeout'` being the test message (from the `test()` function).

Reply:
Fixed